File: src/main/java/info/macias/webapp/security/api/ApiToken.java
package info.macias.webapp.security.api;

import org.apache.shiro.authc.AuthenticationToken;

/**
 * Authentication token for REST API clients that send their credentials by means of an HTTP header
 *
 * @author Mario Macias (http://github.com/mariomac)
 */
public class ApiToken implements AuthenticationToken {

    private String remoteKeyUUID;

    /**
     * Instantiates the class from a remote API key value
     * @param remoteKeyUUID
     */
    public ApiToken(String remoteKeyUUID) {
        this.remoteKeyUUID = remoteKeyUUID;
    }

    public final String getRemoteKeyUUID() {
        return remoteKeyUUID;
    }

    /**
     * Returns the same as getRemoteKeyUUID
     * @return
     */
    @Override
    public Object getPrincipal() {
        return getRemoteKeyUUID();
    }

    /**
     * Returns the same as getRemoteKeyUUID
     * @return
     */
    @Override
    public Object getCredentials() {
        return getRemoteKeyUUID();
    }
}

File: src/main/java/info/macias/webapp/security/api/ApiToken.kt
package info.macias.webapp.security.api

import org.apache.shiro.authc.AuthenticationToken

/**
 * Authentication token for REST API clients that send their credentials by means of an HTTP header

 * @author Mario Macias (http://github.com/mariomac)
 */
class ApiToken
/**
 * Instantiates the class from a remote API key value
 * @param remoteKeyUUID
 */
(val remoteKeyUUID: String) : AuthenticationToken {

    /**
     * Returns the same as getRemoteKeyUUID
     * @return
     */
    override fun getPrincipal(): Any {
        return remoteKeyUUID
    }

    /**
     * Returns the same as getRemoteKeyUUID
     * @return
     */
    override fun getCredentials(): Any {
        return remoteKeyUUID
    }
}
